app/services/xccdf_report_parser.rb

```diff
@@ -45,10 +47,23 @@ def check_os_version
     # rubocop:enable Style/GuardClause
   end
 
+  def check_for_external_reports
+    # rubocop:disable Style/GuardClause
+    if external_report?
+      raise ExternalReportError,
+            "No policy found matching benchmark #{benchmark.ref_id} "\
+            "(RHEL-#{benchmark.os_major_version}) and profile "\
+            "#{test_result_profile.ref_id} with host #{@host.name} assigned "\
+            "for account #{@account.account_number}."
+    end
+    # rubocop:enable Style/GuardClause
+  end
+
   def save_all
     Host.transaction do
       save_host
       check_os_version
+      check_for_external_reports unless Settings.features.parse_external_reports
```

Comment:
I wanted to be sure we were able to detect external reports before saving any benchmark info. If the benchmark isn't already saved, it will still work; however, this case will stop being possible once we stop parsing reports from unknown benchmarks in RHICOMPL-722.